packages/vm/src/runBlock.ts

```diff
@@ -951,17 +957,143 @@ const DAOConfig = {
   DAORefundContract: 'bf4ed7b27f1d666546e30d74d50d173d20bca754',
 }
 
+export class ValidatorWithdrawalRequest extends CLRequest implements CLRequestType {
+  constructor(type: number, bytes: Uint8Array) {
+    super(type, bytes)
+  }
+
+  serialize() {
+    return concatBytes(Uint8Array.from([this.type]), this.bytes)
+  }
+}
+
 /**
  * This helper method generates a list of all CL requests that can be included in a pending block
  * @param _vm VM instance from which to derive CL requests
  * @returns an list of CL requests in ascending order by type
  */
-export const accumulateRequests = async (_vm: VM): Promise<CLRequest[]> => {
+export const accumulateRequests = async (vm: VM): Promise<CLRequest[]> => {
   const requests: CLRequest[] = []
+  const common = vm.common
 
   // TODO: Add in code to accumulate deposits (EIP-6110)
 
-  // TODO: Add in code to accumulate partial withdrawals (EIP-7002)
+  if (common.isActivatedEIP(7002)) {
+    // Partial withdrawals logic
+    const addressBytes = setLengthLeft(
+      bigIntToBytes(common.param('vm', 'withdrawalRequestPredeployAddress')),
+      20
+    )
+    const withdrawalsAddress = Address.fromString(bytesToHex(addressBytes))
+
+    const withdrawalsType = Number(common.param('vm', 'withdrawalRequestType'))
+
+    const excessWithdrawalsSlot = setLengthLeft(
+      bigIntToBytes(common.param('vm', 'excessWithdrawalsRequestStorageSlot')),
+      32
+    )
+    const withdrawalsCountSlot = setLengthLeft(
+      bigIntToBytes(common.param('vm', 'withdrawalsRequestCountStorage')),
+      32
+    )
+    const queueHeadSlot = setLengthLeft(
+      bigIntToBytes(common.param('vm', 'withdrawalsRequestQueueHeadStorageSlot')),
+      32
+    )
+    const queueTailSlot = setLengthLeft(
+      bigIntToBytes(common.param('vm', 'withdrawalsRequestTailHeadStorageSlot')),
+      32
+    )
+
+    const maxWithdrawalsPerBlock = common.param('vm', 'maxWithdrawalRequestsPerBlock')
+    const targetWithdrawalRequestsPerBlock = common.param('vm', 'targetWithdrawalRequestsPerBlock')
+    const offset = common.param('vm', 'withdrawalsRequestQueueStorageOffset')
+
+    // Dequeue withdrawal requests
+
+    const queueHeadIndex = bytesToBigInt(
+      await vm.stateManager.getContractStorage(withdrawalsAddress, queueHeadSlot)
```

Comment:
Maybe for user convenience to wrap this first withdrawals system contract access in a try/catch together with a meaningful error msg? („….failed, likely withdrawals system contract not deployed“ or similar)

Reply:
Yes, good point!

Reply:
Have addressed!